<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>King of Knowledge</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>C:/Users/allen/OneDrive/Desktop/asetf-vdw5x-001.ico</normaloff>C:/Users/allen/OneDrive/Desktop/asetf-vdw5x-001.ico</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>731</width>
      <height>511</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <widget class="QWidget" name="home_page">
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <widget class="QPushButton" name="gameStart_btn">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>120</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>18</pointsize>
       </font>
      </property>
      <property name="text">
       <string>開始遊戲</string>
      </property>
     </widget>
     <widget class="QPushButton" name="about_btn">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>250</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>18</pointsize>
       </font>
      </property>
      <property name="text">
       <string>遊戲介紹</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="about_game_page">
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>70</y>
        <width>241</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="text">
       <string>遊戲介紹</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>220</y>
        <width>441</width>
        <height>141</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>每一局會有一道題目，最快答對的可以得到 10 分，最慢的得到 5 分。時間限制為 10 秒，10 秒內沒有答出來就會換下一題。最後看誰得分最多就是贏家</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>100</y>
        <width>431</width>
        <height>141</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>!!知識王挑戰!!</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="back_to_home_page_btn">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>390</y>
        <width>141</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="text">
       <string>回到首頁</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="connect_page">
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>140</y>
        <width>211</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Ur cool nickname :</string>
      </property>
     </widget>
     <widget class="QPushButton" name="connect_btn">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>330</y>
        <width>191</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>連接伺服器</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>230</y>
        <width>121</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Server IP :</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="client_name">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>160</y>
        <width>241</width>
        <height>41</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="server_ip">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>250</y>
        <width>241</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>127.0.0.1</string>
      </property>
     </widget>
     <widget class="QLabel" name="title">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>581</width>
        <height>101</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="text">
       <string>King of Knowledge</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="back_home_page_btn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>330</y>
        <width>191</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>回到首頁</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="game_page">
     <widget class="QLabel" name="question_lbl">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>0</y>
        <width>651</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>題目</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="choice1_btn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>180</y>
        <width>301</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>選項</string>
      </property>
     </widget>
     <widget class="QPushButton" name="choice4_btn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>420</y>
        <width>301</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>選項</string>
      </property>
     </widget>
     <widget class="QPushButton" name="choice3_btn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>340</y>
        <width>301</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>選項</string>
      </property>
     </widget>
     <widget class="QPushButton" name="choice2_btn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>260</y>
        <width>301</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>選項</string>
      </property>
     </widget>
     <widget class="QLabel" name="player1_name">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>90</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>玩家1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="player2_name">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>90</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>玩家2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QProgressBar" name="player1_score">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>118</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>24</number>
      </property>
      <property name="textVisible">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QProgressBar" name="player2_score">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>140</y>
        <width>118</width>
        <height>23</height>
       </rect>
      </property>
      <property name="value">
       <number>24</number>
      </property>
     </widget>
     <widget class="QLCDNumber" name="timer_lcd">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>100</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <property name="smallDecimalPoint">
       <bool>false</bool>
      </property>
      <property name="segmentStyle">
       <enum>QLCDNumber::Filled</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="gameOver_page">
     <widget class="QLabel" name="game_status">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>180</y>
        <width>591</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="text">
       <string>挑戰成功</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="home_page_btn">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>360</y>
        <width>141</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>回到首頁</string>
      </property>
     </widget>
     <widget class="QLabel" name="player_name1">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>100</y>
        <width>111</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="text">
       <string>玩家1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="player_name2">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>100</y>
        <width>111</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="text">
       <string>玩家2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="rank_page_btn">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>360</y>
        <width>141</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>排行榜</string>
      </property>
     </widget>
     <widget class="QLabel" name="player_score1">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>50</y>
        <width>71</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>得分: </string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="player_score2">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>50</y>
        <width>71</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>得分: </string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="rank_page">
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>120</y>
        <width>301</width>
        <height>301</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>10</horstretch>
        <verstretch>10</verstretch>
       </sizepolicy>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="textElideMode">
       <enum>Qt::ElideMiddle</enum>
      </property>
      <property name="showGrid">
       <bool>true</bool>
      </property>
      <property name="gridStyle">
       <enum>Qt::SolidLine</enum>
      </property>
      <property name="cornerButtonEnabled">
       <bool>true</bool>
      </property>
      <property name="rowCount">
       <number>5</number>
      </property>
      <property name="columnCount">
       <number>3</number>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <column/>
      <column/>
      <column/>
      <item row="0" column="0">
       <property name="text">
        <string>排名</string>
       </property>
       <property name="font">
        <font>
         <family>Microsoft JhengHei UI</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="textAlignment">
        <set>AlignCenter</set>
       </property>
      </item>
      <item row="0" column="1">
       <property name="text">
        <string>玩家</string>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="textAlignment">
        <set>AlignCenter</set>
       </property>
      </item>
      <item row="0" column="2">
       <property name="text">
        <string>勝率</string>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="textAlignment">
        <set>AlignCenter</set>
       </property>
      </item>
      <item row="1" column="0">
       <property name="text">
        <string/>
       </property>
      </item>
      <item row="1" column="1">
       <property name="text">
        <string/>
       </property>
      </item>
      <item row="2" column="1">
       <property name="font">
        <font>
         <strikeout>false</strikeout>
         <stylestrategy>PreferAntialias</stylestrategy>
        </font>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>50</y>
        <width>271</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Rank</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="back_home_btn">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>420</y>
        <width>141</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>微軟正黑體</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>回到首頁</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
